### PR TITLE
Remove compilation warnings

### LINKED ISSUE #97: Symbol’s function definition is void: annotation when some text is selected in buffer
When I select some text and invoke `annotate-annotate` I'm getting this error. Otherwise, things work as expected and the word at point gets used. Using doom, and package installed from MELPA. 

```
annotate-annotations-overlay-in-range: Symbol’s function definition is void: annotation
```

### PR DESCRIPTION
**Pull request by cage2**
Hi!

After trying to address #97 and the excellent suggestion from Emacs Doom maintainer from https://github.com/hlissner/doom-emacs/issues/4783 i realized i introduced many subtle inconsistencies in the code (and even an actual bug as in #97!).

Fortunately the compiler is smart enough ;-) to recognize some of them and i tried to make it happy removing al the warning (and hopefully resolve #97 as well). :)

Bye!
C.

### CODE REVIEW

File: annotate.el [hunk 1]
```diff
@@ -424,19 +424,19 @@ modified (for example a newline is inserted)."
...
                  (annotate--remap-chain-pos chain)
-                 (font-lock-ensure))))))))))
+                 (font-lock-flush))))))))))
 
 (defun annotate-info-select-fn ()
   "The function to be called when an info buffer is updated"
   (annotate-clear-annotations)
   (annotate-load-annotations)
-  (font-lock-ensure nil))
+  (font-lock-flush nil))
```
Comment (bastibe) [inline annotate.el]: The nil argument here is probably unnecessary now.

**Comment by bastibe** [comment]
Brilliant! It's so great to see Emacs folks all over the planet helping one another.

**Comment by cage2** [comment]
On Sun, Mar 14, 2021 at 05:49:25AM -0700, Bastian Bechtold wrote:

Hi!

> Brilliant! It's so great to see Emacs folks all over the planet helping one another.

And thanks  to you for carefully  check the code i  write (something i
really need)!

Bye!
C.

**Comment by cage2** [comment]
On Sun, Mar 14, 2021 at 05:48:12AM -0700, Bastian Bechtold wrote:
> -  (font-lock-ensure nil))
> +  (font-lock-flush nil))
>
> The nil argument here is probably unnecessary now.

True! Just pushed the fix.

Bye!
C.

**Comment by bastibe** [comment]
Quite a big new release! Thank you for all your hard work!

**Comment by cage2** [comment]
On Sun, Mar 21, 2021 at 01:12:52PM -0700, Bastian Bechtold wrote:

Hi!

> Quite a big new release! Thank you for all your hard work!

Thank you!   I feel  a bit  sad that i  did not  tried to  compile the
package before this way  i left some bug unnoticed (by  me), i think i
have learnt the lesson and i am  going to compile the package to check
for warnings.

Bye!
C.

**Comment by cage2** [comment]
Merging, thanks to all the people that helped with this patch!

**Pull request merged**